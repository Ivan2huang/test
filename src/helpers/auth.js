export const getCookie = name => {
  const value = `; ${document.cookie}`;
  const parts = value.split(`; ${name}=`);
  return parts
    .pop()
    .split(';')
    .shift();
};

export const hasCookie = name => {
  const value = getCookie(name);
  return value && value.trim().length > 0;
};
