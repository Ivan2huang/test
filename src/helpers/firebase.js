import firebase from 'firebase/app';
import 'firebase/analytics';
import 'firebase/remote-config';
import CONFIG from '../constants/config';
import logger from './logger';

const config = {
  apiKey: CONFIG.firebaseApiKey,
  authDomain: CONFIG.firebaseAuthDomain,
  databaseURL: CONFIG.firebaseDatabaseUrl,
  projectId: CONFIG.firebaseProjectId,
  storageBucket: CONFIG.firebaseStorageBucket,
  messagingSenderId: CONFIG.firebaseMessagingSenderId,
  appId: CONFIG.firebaseAppId,
  measurementId: CONFIG.firebaseMeasurementId,
};

let firebaseApp;

export const initFirebase = () => {
  if (!firebaseApp) {
    firebaseApp = firebase.initializeApp(config);
  } else {
    firebaseApp = firebase.app();
  }
};

export const trackUserInfo = ({ clientId, memberId, role }) => {
  const analytics = firebaseApp.analytics();
  analytics.setUserId(memberId);
  analytics.setUserProperties({
    client_id: clientId,
    user_type: role,
  });
};

export const logEvent = (eventName, category, action, label, options = {}) => {
  if (CONFIG.enableGA) {
    firebaseApp.analytics().logEvent(eventName, {
      category,
      action,
      label,
      ...options,
    });
  }
};

export const logAction = ({ category, action, ...restParams }) => {
  if (CONFIG.enableGA) {
    firebaseApp.analytics().logEvent('user_action', {
      category,
      action,
      ...restParams,
    });
  }
};

export const trackingAnchor = e => {
  e.preventDefault();
  const anchorNode = e.target;
  const href = anchorNode.getAttribute('href');
  const { category, action } = anchorNode.dataset;
  logAction({
    category,
    action,
  });
  window.location.assign(href);
};

export const getFirebaseApp = () => firebaseApp;

export const getRemoteFeatureChecker = async ({ parameterName }) => {
  if (!firebaseApp) {
    return () => false;
  }

  const remoteConfig = firebaseApp.remoteConfig();
  remoteConfig.settings = {
    minimumFetchIntervalMillis: 3600000,
  };
  let featureMap = {};
  try {
    await remoteConfig.fetchAndActivate();
    const remoteValues = await remoteConfig.getAll();
    if (remoteValues && remoteValues[parameterName]) {
      featureMap = JSON.parse(remoteValues[parameterName].asString());
    }
  } catch (e) {
    logger.error('---getRemoteFeatureChecker -> ERROR---', e);
  }
  return featureName => {
    return featureMap[featureName];
  };
};
