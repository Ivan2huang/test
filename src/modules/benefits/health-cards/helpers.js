/* eslint-disable import/prefer-default-export */
export const getQRCodeXML = ({
  fullName,
  membershipNumber,
  certificateNumber,
  insurerCode,
  insurerName,
  insurerType,
  policyNumber,
  expiryDate,
}) =>
  [
    '<?xml version="1.0"?>',
    '<medical_card>',
    '<Card_type>',
    insurerType,
    '</Card_type>',
    '<insurance_company_code>',
    insurerCode,
    '</insurance_company_code>',
    '<insurance_company_name>',
    insurerName,
    '</insurance_company_name>',
    '<insured_name>',
    fullName,
    '</insured_name>',
    '<policy_no>',
    policyNumber,
    '</policy_no>',
    '<certificate_no>',
    certificateNumber,
    '</certificate_no>',
    '<membership_no>',
    membershipNumber,
    '</membership_no>',
    '<expiry_date>',
    expiryDate,
    '</expiry_date>',
    '</medical_card>',
  ].join('');
