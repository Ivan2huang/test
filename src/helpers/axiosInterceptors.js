/* eslint-disable import/no-cycle */
import url from './url';
import paths from './paths';
import { error as errorCodes } from '../constants/auth';

let requestsToRefresh = [];
let isRefreshRequesting = false;

export default axios => {
  if (axios) {
    axios.interceptors.response.use(
      response => {
        return response;
      },
      err => {
        const { response, config } = err;

        if ([401, 403].indexOf(response.status) !== -1) {
          if (!isRefreshRequesting) {
            isRefreshRequesting = true;
            axios
              .get(url.refreshToken)
              .then(({ data }) => {
                const { isSuccess, error } = data;
                if (isSuccess) {
                  requestsToRefresh.forEach(cb => cb(true));
                } else {
                  const redirectUrl =
                    error === errorCodes.token.invalidRefreshToken
                      ? paths.common.tokenExpired
                      : paths.common.login;
                  window.location.replace(redirectUrl);
                }
              })
              .catch(() => {
                requestsToRefresh.forEach(cb => cb(false));
              })
              .finally(() => {
                requestsToRefresh = [];
                isRefreshRequesting = false;
              });
          }

          return new Promise((resolve, reject) => {
            requestsToRefresh.push(hasToken => {
              if (hasToken) {
                resolve(axios(config));
              } else {
                reject(Promise.reject(err));
              }
            });
          });
        }

        return Promise.reject(err);
      },
    );
  }
};
