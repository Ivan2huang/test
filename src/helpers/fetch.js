/* eslint-disable import/no-cycle */
/* eslint-disable no-console,consistent-return */

import axios from 'axios';
import { navigateTo } from './helpers';
import paths from './paths';
import useAxiosInterceptors from './axiosInterceptors';
import getLocale from '../i18n/getLocale';
import CONFIG from '../constants/config';
import logger from './logger';

if (CONFIG.enableRefreshToken) {
  useAxiosInterceptors(axios);
}

export const ContentTypes = {
  JSON: 'application/json;charset=UTF-8',
  FILE: 'application/x-www-form-urlencoded',
};

export const defaultHeader = () => ({
  accept: ContentTypes.JSON,
  'Content-Type': ContentTypes.JSON,
  'Accept-Language': getLocale(),
  Pragma: 'no-cache',
});

const uploadFileHeader = () => ({
  'Content-Type': ContentTypes.FILE,
});

const handleError = (error, expectsErrorResponse) => {
  let response = false;
  logger.error('fetch -> ERROR', error);
  if (error.response && error.response.data) {
    response =
      error.response.data.errors && error.response.data.errors.length >= 1
        ? {
            error: true,
            ...error.response.data.errors[0],
          }
        : {
            error: true,
            messageKey: error.response.data.code,
            message: error.response.data.message,
          };
  } else if (error && !error.status) {
    // network error
    response = {
      error: true,
      messageKey: 'NETWORK_ERROR',
      message: '',
    };
  }

  if (expectsErrorResponse) {
    return response;
  }

  if (error.response && error.response.status === 404) {
    navigateTo(paths.common.notfound);
  } else {
    navigateTo(paths.common.error);
  }
};

export const fetchData = (
  method,
  url,
  body,
  expectsErrorResponse = false,
  headerOptions = {},
) => {
  return axios(url, {
    headers: {
      ...defaultHeader(),
      ...headerOptions,
    },
    method,
    data: JSON.stringify(body),
  })
    .then(response => response.data)
    .catch(error => handleError(error, expectsErrorResponse));
};

export const uploadFile = (method, url, body, expectsErrorResponse = false) => {
  return axios(url, {
    headers: uploadFileHeader(),
    method,
    data: body,
  })
    .then(response => response.data)
    .catch(error => handleError(error, expectsErrorResponse));
};

export const fetchFile = url => {
  const axiosOption = { responseType: 'blob' };
  return axios(url, axiosOption)
    .then(response => response.data)
    .catch(error => handleError(error, false));
};
