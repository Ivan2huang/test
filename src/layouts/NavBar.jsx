/* eslint-disable jsx-a11y/no-autofocus */

import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { withRouter } from 'next/router';

import { Tabs, Hidden, Tooltip, withStyles } from '@material-ui/core';
import { injectIntl } from 'react-intl';
import TrackingTab from '../uiComponents/TrackingTab';
import paths from '../helpers/paths';
import NAV_TABS from './navTabs';
import { navigateTo, convertObjectToParam } from '../helpers/helpers';
import { CATEGORIES } from '../constants/analytics';

const styles = ({
  headerNavLink,
  headerNavLinkActive,
  disabled,
  white,
  spacingX,
}) => ({
  flexContainer: {
    height: spacingX(20),
    paddingRight: spacingX(10),
  },
  tabRoot: {
    marginLeft: spacingX(13),
    color: headerNavLink || disabled,
    fontSize: '14px',
    fontWeight: 400,

    '&$selected': {
      fontWeight: 400,
      color: headerNavLinkActive || white,
    },
  },
  selected: {},
});

const getTooltips = (intl, nav) => (
  <Tooltip title={nav.displayName} placement="bottom-end">
    <span>
      {intl.formatMessage({
        id: `nav-${nav.name}`,
        defaultMessage: nav.displayName,
      })}
    </span>
  </Tooltip>
);

const NavBar = ({ router, classes, intl }) => {
  const selectedTabIndex = NAV_TABS.findIndex(tab =>
    router.pathname.startsWith(paths.common[tab.name]),
  );

  const setSelectedTab = (_, newSelectedTabIndex) => {
    const tab = NAV_TABS[newSelectedTabIndex];
    let params;
    if (tab.getParams) {
      params = tab.getParams();
    }
    if (tab.externalUrl) {
      window.open(
        `${tab.externalUrl}?${convertObjectToParam(params)}`,
        '_self',
      );
    } else {
      navigateTo(paths.common[tab.name], params);
    }
  };

  return (
    <Hidden implementation="css" xsDown>
      <Tabs
        value={selectedTabIndex}
        onChange={setSelectedTab}
        indicatorColor="primary"
        classes={{
          flexContainer: classes.flexContainer,
        }}
      >
        {NAV_TABS.map(nav => (
          <TrackingTab
            disableRipple
            disableFocusRipple
            key={`nav-nav-${nav.name}`}
            data-testid={`nav-${nav.name}`}
            label={getTooltips(intl, nav)}
            classes={{ root: classes.tabRoot, selected: classes.selected }}
            trackingData={{
              category: CATEGORIES.NAVIGATION_BAR,
              action: nav.trackingName,
            }}
          />
        ))}
      </Tabs>
    </Hidden>
  );
};

NavBar.propTypes = {
  router: PropTypes.shape({}).isRequired,
  intl: PropTypes.shape({}).isRequired,
  classes: PropTypes.exact({
    flexContainer: PropTypes.string.isRequired,
    tabRoot: PropTypes.string.isRequired,
    selected: PropTypes.string.isRequired,
  }).isRequired,
};

export default compose(
  injectIntl,
  withRouter,
  withStyles(styles),
)(NavBar);
