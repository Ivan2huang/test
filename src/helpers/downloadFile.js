import { fetchFile } from './fetch';

const SUCCESS = 'SUCCESS';
const ERROR = 'ERROR';

const downloadFileAsync = async (url, fileName) => {
  const response = await fetchFile(url);
  if (response) {
    const blob = new Blob([response]);
    if (window.navigator.msSaveOrOpenBlob) {
      window.navigator.msSaveOrOpenBlob(blob, fileName);
      return SUCCESS;
    }
    const objectUrl = window.URL.createObjectURL(blob);
    const link = document.createElement('a');
    link.href = objectUrl;
    link.setAttribute('download', fileName);
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
    return SUCCESS;
  }
  return ERROR;
};

export default downloadFileAsync;
