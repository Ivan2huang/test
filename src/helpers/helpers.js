import Router from 'next/router';

import zhLocale from 'date-fns/locale/zh-CN';
import enLocale from 'date-fns/locale/en-US';
import * as Sentry from '@sentry/node';

import DateFnsUtils from '@date-io/date-fns';
import { getCookie } from './auth';
import { USER_ID } from '../constants/auth';
import CONFIG from '../constants/config';
// eslint-disable-next-line import/no-cycle
import { fetchFile } from './fetch';

const PRECISION = 2;

export const LOCALES_MAP = {
  'en-HK': enLocale,
  'zh-HK': zhLocale,
};

export const formatMessage = (intl, id, defaultMessage, values) =>
  intl.formatMessage({ id, defaultMessage }, values);

export const maskEmail = email => {
  const [localPart, domain] = email.split('@');
  return `${localPart[0]}******${localPart[localPart.length - 1]}@${domain}`;
};

export const formatAmount = (intl, amount) =>
  intl.formatNumber(amount, {
    minimumFractionDigits: PRECISION,
    maximumFractionDigits: PRECISION,
  });

export const navigateTo = (path, queryParam = {}, asPath) => {
  console.log('helpers.js navigateTo path=', path);
  Router.push({ pathname: path, query: queryParam }, asPath);
};

export const buildClaimPostBody = (
  formData,
  receiptFilesIds,
  referralFileId,
) => {
  const selectedPatientId = formData.patient.patientId;
  const userId = getCookie(USER_ID);
  const claimantName =
    selectedPatientId === userId
      ? `Employee|${selectedPatientId}`
      : `Dependant|${selectedPatientId}`;

  return {
    benefitPeriod: formData.planId,
    claimantName,
    receiptDate: formData.claim.consultationDate,
    receiptAmount: formData.claim.receiptAmount,
    claimReason: formData.claim.diagnosis,
    claimTypeId: formData.claim.claimId,
    claimItemName: formData.claimType,
    int1: true,
    str5: formData.patient.contactNumber,
    receiptFilesIds,
    referralFileId,
    int2: formData.claim.anotherInsurer ? 1 : 0,
    dec1: formData.claim.otherInsurerAmount,
  };
};

const isValidFileType = (validFileType, fileName) =>
  validFileType.includes(
    fileName
      .split('.')
      .pop()
      .toLowerCase(),
  );

const isValidFileSize = (validFileSize, fileSize) => fileSize <= validFileSize;

export const validateFiles = (files, validFileType, validFileSize) => {
  return files.filter(
    ({ name, size }) =>
      isValidFileType(validFileType, name) &&
      isValidFileSize(validFileSize, size),
  );
};

export const isEmpty = obj =>
  Object.entries(obj).length === 0 && obj.constructor === Object;

export const today = () =>
  new DateFnsUtils().format(Date.now(), CONFIG.dateFormat);

export const formatDate = (
  date,
  locale = null,
  customFormat = CONFIG.dateFormat,
) => {
  try {
    const dateFnsUtil = new DateFnsUtils();
    if (date && date.includes('0001')) {
      return undefined;
    }
    if (locale) {
      dateFnsUtil.locale =
        LOCALES_MAP[locale] || LOCALES_MAP[CONFIG.locales.ENGLISH];
    }
    return dateFnsUtil.format(new Date(date), customFormat);
  } catch (e) {
    return undefined;
  }
};

export const getCurrentDate = () => new Date();

export const addDays = days => new DateFnsUtils().addDays(new Date(), days);

export const openFile = async (name, url, preloaded = true) => {
  if (window.navigator && window.navigator.msSaveOrOpenBlob) {
    if (!preloaded) {
      window.open(url);
      return;
    }

    const blob = await fetchFile(url);
    window.navigator.msSaveOrOpenBlob(blob, name);
    return;
  }

  window.open(url);
};
export const onEnter = (event, callback) => {
  if (callback && event.key === 'Enter') {
    callback();
  }
};

export const saveToLocalStorage = (key, value) => {
  window.localStorage.setItem(key, value);
};

export const getValueFromLocalStorage = key => {
  return window.localStorage.getItem(key);
};

export const sentenceCase = input => {
  return input.replace(/\w\S*/g, word => {
    return word.charAt(0).toUpperCase() + word.substr(1).toLowerCase();
  });
};

export const delay = waitFor =>
  new Promise(resolve => {
    setTimeout(() => resolve(), waitFor);
  });

export const objectToURLParams = obj =>
  Object.keys(obj)
    .map(key => {
      return `${key}=${encodeURIComponent(obj[key])}`;
    })
    .join('&');

export const snakeToCamel = str =>
  str.replace(/([-_]\w)/g, g => g[1].toUpperCase());

export const objectKeySnakeToCamel = obj => {
  return Object.entries(obj).reduce((result, [key, value]) => {
    return {
      ...result,
      [snakeToCamel(key)]: value,
    };
  }, {});
};

export const isValidLanguageCode = code => {
  return (CONFIG.supportedLanguages || []).includes(code);
};

/* istanbul ignore next */
export const captureException = (error, errorInfo = {}) => {
  Sentry.withScope(scope => {
    Object.keys(errorInfo).forEach(key => {
      scope.setExtra(key, errorInfo[key]);
    });

    if (error.response && error.response.data) {
      scope.setExtra(
        'requestErrors',
        JSON.stringify(error.response.data.errors || []),
      );
    }

    Sentry.captureException(error);
  });
};

export const removeUnicodeScript = str => {
  if (!str) return '';
  const pattern = /[\u006E\u00B0\u00B2\u00B3\u00B9\u02AF\u0670\u0711\u2121\u213B\u2207\u29B5\uFC5B-\uFC5D\uFC63\uFC90\uFCD9\u2070\u2071\u2074-\u208E\u2090-\u209C\u0345\u0656\u17D2\u1D62-\u1D6A\u2A27\u2C7C]+\s/;
  return str.replace(pattern, '').trim();
};

export const convertObjectToParam = (
  obj,
  compareSign = '=',
  delimiter = '&',
) => {
  if (typeof obj === 'object') {
    return Object.keys(obj)
      .map(key => `${key}${compareSign}${obj[key]}`)
      .join(delimiter);
  }
  return '';
};

export const lowercaseObjectKeys = obj => {
  if (typeof obj !== 'object') return {};

  const result = {};
  Object.keys(obj).forEach(key => {
    result[key.toLowerCase()] = obj[key];
  });

  return result;
};
