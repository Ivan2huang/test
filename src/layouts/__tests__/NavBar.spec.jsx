/* eslint-disable react/prop-types */

import React from 'react';
import { render, fireEvent, wait } from '@testing-library/react';

import NavBar from '../NavBar';
import withTheme from '../../themes/withThemeProvider';
import { navigateTo } from '../../helpers/helpers';
import withIntl from '../../i18n/withIntlProvider';
import CONFIG from '../../constants/config';

jest.mock('../../helpers/auth', () => ({
  getCookie: jest.fn().mockReturnValueOnce('clientId'),
}));

jest.mock('../../helpers/helpers', () => ({
  navigateTo: jest.fn(),
  convertObjectToParam: jest.fn().mockReturnValueOnce('clientId=clientId'),
}));
jest.mock('../../helpers/firebase', () => ({
  logEvent: jest.fn(),
  logAction: jest.fn(),
}));

jest.mock('@material-ui/core/Hidden', () => ({ children, smUp }) => (
  <div data-component="MuiHidden" data-smup={smUp}>
    {children}
  </div>
));

jest.mock('@material-ui/core/Tooltip', () => ({ children }) => <>{children}</>);

jest.mock('../../i18n/getLocale', () => () => 'en');

jest.mock('../../i18n/lang', () => ({
  messages: {},
  langs: ['en'],
}));

jest.mock('../../constants/config', () => ({
  shopUrl: 'shop.com',
  defaultLanguage: 'en',
}));

describe('NavBar Component', () => {
  beforeEach(() => {
    navigateTo.mockClear();
  });

  it('should match snapshot', () => {
    const props = {
      router: {
        pathname: '/claims',
      },
    };
    const Component = withIntl(withTheme(NavBar));

    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should navigate to link', async () => {
    const props = {
      router: {
        pathname: '/lifestyle',
      },
    };

    const Component = withIntl(withTheme(NavBar));

    const { getByTestId } = render(<Component {...props} />);
    const claimsNav = getByTestId('nav-claims');
    fireEvent.click(claimsNav);

    await wait(() => {
      expect(navigateTo).toHaveBeenCalledTimes(1);
      expect(navigateTo).toHaveBeenCalledWith('/claims', undefined);
    });
  });

  it('should navigate to link with params', async () => {
    const props = {
      router: {
        pathname: '/claims',
      },
    };

    const Component = withIntl(withTheme(NavBar));

    const { getByTestId } = render(<Component {...props} />);
    const claimsNav = getByTestId('nav-lifestyle');
    fireEvent.click(claimsNav);

    await wait(() => {
      expect(navigateTo).toHaveBeenCalledTimes(1);
      expect(navigateTo).toHaveBeenCalledWith('/lifestyle', {
        clientId: 'clientId',
      });
    });
  });

  it('should navigate to external Url if available', () => {
    window.open = jest.fn();
    const props = {
      router: {
        pathname: '/eshop',
      },
    };
    const Component = withIntl(withTheme(NavBar));
    const { getByTestId } = render(<Component {...props} />);
    const shopNav = getByTestId('nav-eshop');

    fireEvent.click(shopNav);

    expect(window.open).toHaveBeenCalledWith(
      `${CONFIG.shopUrl}?clientId=clientId`,
      '_self',
    );
    expect(window.open).toHaveBeenCalledTimes(1);
  });
});
