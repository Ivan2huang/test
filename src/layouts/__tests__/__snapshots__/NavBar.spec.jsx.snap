// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should match snapshot 1`] = `
<div>
  <div
    data-component="MuiHidden"
  >
    <div
      class="MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer NavBar-flexContainer-1"
          role="tablist"
        >
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root NavBar-tabRoot-2 MuiTab-textColorInherit"
            data-testid="nav-lifestyle"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <span>
                Lifestyle
              </span>
            </span>
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root NavBar-tabRoot-2 MuiTab-textColorInherit"
            data-testid="nav-doctors"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <span>
                Doctors
              </span>
            </span>
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root NavBar-tabRoot-2 MuiTab-textColorInherit"
            data-testid="nav-benefits"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <span>
                Benefits
              </span>
            </span>
          </button>
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root NavBar-tabRoot-2 MuiTab-textColorInherit Mui-selected NavBar-selected-3"
            data-testid="nav-claims"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <span>
                Claims
              </span>
            </span>
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root NavBar-tabRoot-2 MuiTab-textColorInherit"
            data-testid="nav-eshop"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <span>
                Choices
              </span>
            </span>
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root NavBar-tabRoot-2 MuiTab-textColorInherit"
            data-testid="nav-me"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              <span>
                Me
              </span>
            </span>
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-28 PrivateTabIndicator-colorPrimary-29 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;
